version: "3.8"

services:
  test:
    build: .
    container_name: warehouse_tests
    command: ["pytest", "--disable-warnings"]
    depends_on:
      test_db:
        condition: service_healthy
    env_file:
      - .env.test
    environment:
      DB_HOST: test_db
    volumes:
      - .:/app
    working_dir: /app
    restart: "no"

  test_db:
    image: postgres:15
    container_name: warehouse_test_db
    env_file:
      - .env.test
    ports:
      - "5433:5432"
    volumes:
      - pgdata_test:/var/lib/postgresql/data
    restart: always
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_SUPERUSER} -d ${DB_NAME}"]
      interval: 5s
      timeout: 5s
      retries: 3
    environment:
      POSTGRES_USER: ${POSTGRES_SUPERUSER}
      POSTGRES_PASSWORD: ${POSTGRES_SUPERUSER_PASSWORD}

volumes:
  pgdata_test:
